test/http.js

```diff
@@ -360,64 +359,55 @@ describe('http', () => {
-        const headers = {
-          Authorization: ['Basic hoop-la', 'Advanced hoop-la']
-        }
-
-        const res = fetchMock.mock('http://swagger.io', {headers})
+        const response = new Response('data', {
+          status: 200,
+          headers: {
+            Authorization: 'Basic hoop-la, Advanced hoop-la',
+            'Content-Type': 'text/plain',
+          }
+        })
 
-        return fetch('http://swagger.io').then((_res) => { // eslint-disable-line no-undef
-          return serializeRes(_res, 'https://swagger.io')
-        }).then((resSerialize) => {
-          expect(resSerialize.headers).toEqual({authorization: ['Basic hoop-la', 'Advanced hoop-la']})
-        }).then(fetchMock.restore)
+        return serializeRes(response, 'https://swagger.io')
+          .then((serializedResponse) => {
+            expect(serializedResponse.headers).toEqual({
+              authorization: ['Basic hoop-la', 'Advanced hoop-la'],
+              'content-type': 'text/plain'
+            })
+          })
       }
     )
 
     test(
       'should set .text and .data to body Blob or Buffer for binary response',
       () => {
-        const headers = {
-          'Content-Type': 'application/octet-stream'
-        }
-
-        const body = 'body data'
-        const res = fetchMock.mock('http://swagger.io', {body, headers})
-
-        let originalRes
-
-        return fetch('http://swagger.io').then((_res) => { // eslint-disable-line no-undef
-          originalRes = _res
-          return serializeRes(_res, 'https://swagger.io')
-        }).then((resSerialize) => {
-          expect(resSerialize.data).toBe(resSerialize.text)
-          if (originalRes.blob) {
-            expect(resSerialize.data).toBeInstanceOf(Blob) // eslint-disable-line no-undef
+        const response = new Response('body data', {
+          headers: {
+            'Content-Type': 'application/octet-stream'
           }
-          else {
-            expect(resSerialize.data).toBeInstanceOf(Buffer)
-            expect(resSerialize.data).toEqual(new Buffer(body))
-          }
-        }).then(fetchMock.restore)
+        })
+
+        return serializeRes(response, 'https://swagger.io')
+          .then((serializedResponse) => {
+            expect(serializedResponse.data).toBe(serializedResponse.text)
+            expect(Object.prototype.toString.call(serializedResponse.data)).toBe('[object Blob]')
+          })
       }
     )
 
     test(
       'should set .text and .data to body string for text response',
       () => {
-        const headers = {
-          'Content-Type': 'application/json'
-        }
-
-        const body = 'body data'
-        const res = fetchMock.mock('http://swagger.io', {body, headers})
+        const response = new Response('body data', {
+          headers: {
+            'Content-Type': 'application/json'
+          }
+        })
 
-        return fetch('http://swagger.io').then((_res) => { // eslint-disable-line no-undef
```

Comment:
I think we should remove the fetchMock test, and add a separate non-fetchMock test.

Reply:
Agree, returned back the tests and fixed their underlying fundamental issue of fetchMock using different version of node-fetch as is used in production build. More [here](https://github.com/swagger-api/swagger-js/pull/1523/files#diff-bd8228415b2a4b7ae1c42e201e63da4b).

Reply:
Actual header serialization test on multiple values is [here](https://github.com/swagger-api/swagger-js/blob/80f9bf1bb933b60949b8829d94dd3f827d9c40f9/test/http.js#L210).